lib/ash/actions/read/relationships.ex

```diff
@@ -1191,6 +1251,9 @@ defmodule Ash.Actions.Read.Relationships do
       has_page? = query.page not in [nil, false]
 
       cond do
+        is_list(relationship.through) ->
+          false
```

Comment:
So this will actually be the way that we enable data layers to join through relationships. So this should return `true` if the data layer can `:lateral_join_through_relationships` or something. For now, nothing will implement this, so we can leave it as a stub and talk about that implementation later. It is actually quite complex to do 😄

Reply:
In fact...yeah returning `false` makes sense here, never mind. When its time to support doing this in an optimized way in the data layer, *then* we'd return `true` here.